packages/onboarding/src/cart/index.tsx

```diff
@@ -0,0 +1,317 @@
+		options: {
+			designType: '',
+			theme: themeSlugWithRepo,
+			use_theme_annotation: useThemeHeadstart,
+			default_annotation_as_primary_fallback: true,
+			site_segment: undefined,
+			site_information: {
+				title: siteTitle,
+			},
+			site_creation_flow: flowToCheck,
+			timezone_string: guessTimezone(),
+			wpcom_public_coming_soon: comingSoon,
+			...( siteAccentColor && { site_accent_color: siteAccentColor } ),
+		},
+		validate: false,
+	};
+
+	return newSiteParams;
+};
+
+export const createSiteWithCart = async (
+	flowName: string,
+	isManageSiteFlow: boolean,
+	userIsLoggedIn: boolean,
+	isPurchasingDomainItem: boolean,
+	themeSlugWithRepo: string,
+	comingSoon: 1 | 0,
+	siteTitle: string,
+	siteAccentColor: string,
+	useThemeHeadstart: boolean,
+	productsList: Record< string, ProductListItem >,
+	domainItem?: CartItem
+) => {
+	const siteUrl = domainItem?.meta;
+
+	const newCartItems = [ domainItem ].filter( ( item ) => item );
+	const isFreeThemePreselected = startsWith( themeSlugWithRepo, 'pub' );
+	// x	const bearerToken = get( getSignupDependencyStore( state ), 'bearer_token', null );
+
+	if ( isManageSiteFlow ) {
+		const siteSlugManaged = getSignupCompleteSlug();
+		const newCartItems = [ domainItem ].filter( ( item ) => item );
+
+		await processItemCart(
+			newCartItems,
+			siteSlugManaged,
+			isFreeThemePreselected,
+			themeSlugWithRepo,
+			flowName,
+			userIsLoggedIn,
+			productsList
+		);
+		return;
+	}
+
+	const newSiteParams = getNewSiteParams( {
+		flowToCheck: flowName,
+		isPurchasingDomainItem,
+		themeSlugWithRepo,
+		siteUrl,
+		siteTitle,
+		siteAccentColor,
+		useThemeHeadstart,
+		comingSoon,
+	} );
+
+	// if ( isEmpty( bearerToken ) && 'onboarding-registrationless' === flowToCheck ) {
+	// 	saveToLocalStorageAndProceed( state, domainItem, themeItem, newSiteParams, callback );
+	// 	return;
+	// }
+
+	const locale = getLocaleSlug();
+
+	const siteCreationResponse: NewSiteSuccessResponse = await wpcom.req.post( '/sites/new', {
+		...newSiteParams,
+		locale,
+		lang_id: getLanguage( locale as string )?.value,
+		client_id: config( 'wpcom_signup_id' ),
+		client_secret: config( 'wpcom_signup_key' ),
+	} );
+
+	if ( ! siteCreationResponse.success ) {
+		// TODO ebuccelli: Manage siteCreationResponse.errors
+		return;
+	}
+
+	const parsedBlogURL = getUrlParts( siteCreationResponse?.blog_details.url );
+	const siteSlug = parsedBlogURL.hostname;
+	const siteId = siteCreationResponse?.blog_details.blogid;
+	const providedDependencies = {
+		siteId,
+		siteSlug,
+		domainItem,
+	};
+
+	if ( isTailoredSignupFlow( flowName ) ) {
+		await setupSiteAfterCreation( { siteId, flowName: flowName } );
+	}
+
+	await processItemCart(
+		newCartItems,
+		siteSlug,
+		isFreeThemePreselected,
+		themeSlugWithRepo,
+		flowName,
+		userIsLoggedIn,
+		productsList
+	);
+
+	return providedDependencies;
+};
+
+function prepareItemForAddingToCart( item: CartItem, lastKnownFlow?: string ) {
```

Comment:
I think that this file could probably also use `MinimalRequestCartProduct` instead of `CartItem`.